File: IN1010/trix/go/sammenlikn/katt.go
package main

type Katt struct {
	alder int
	navn string
}

func (k *Katt) Sammenlikn(k2 *Katt) int {
	if k.alder > k2.alder {
		return 1
	} else if k.alder < k2.alder {
		return -1
	} else {
		return 0
	}
}

func (k *Katt) HentAlder() int {
	return k.alder
}

File: IN1010/trix/06/comparable/Katt.java
import java.util.*;

class Katt implements Comparable<Katt> {
    String navn;
    int alder;

    public Katt(String n, int a) {
        this.navn = n;
        this.alder = a;
    }

    public String toString() {
        return String.format("Navn: %s, alder: %d", this.navn, this.alder);
    }

    @Override
    public int compareTo(Katt other) {
        if (this.alder < other.alder) {
            return -1;
        } else {
            return 1;
        }
    }
}
